wled00/FX_fcn.cpp

```diff
@@ -739,7 +754,12 @@ void IRAM_ATTR Segment::setPixelColor(int i, uint32_t col)
     col = RGBW32(r, g, b, w);
   }
 
+  setPixelColorExact(i, col);
+}
+
+void Segment::setPixelColorExact(int i, uint32_t col) {
```

Comment:
I do not think introduction of new function is necessary.

Reply:
The reason I added that function is because I call it above, but also while rendering the transitions. The `setPixelColor()` function corrects the color for brightness first before setting the pixel in the strip. However, when calculating transitions, that correction has already happened, since we have called `setPixelColor()`, either storing the corrected color in the strip or in the buffer. So when I start moving pixels around in the transition, I have to use the already-corrected color, which is what `setPixelColorExact()` does.

Reply:
Effect will do something like this:
```c++
uint32_t col1 = SEGCOLOR(1);
uint32_t col0 = SEGMENT.color_from_palette(palEntry, false, PALETTE_SOLID_WRAP, 0);
for (unsigned i=0; i<SEGLEN; i++) SEGMENT.setPixelColor(i, color_blend(col1, col0, progress);
```
or
```c++
SEGMENT.setPixelColor(i, color_blend(SEGCOLOR(1), SEGMENT.color_from_palette(palEntry, false, PALETTE_SOLID_WRAP, 0), progress));
```
And from this snippet you can immediately see where the problem lies. `SEGCOLOR()` is unaware of the pixel that is being retrieved. Same goes for `SEGMENT.color_from_palette()`. The only place where relation to pixel index happens is in `Segment::setPixelColor()` and `setPixelColor()` receives the exact color that it needs to paint.
Unfortunately you do not know if this color is from *color slot* (`SEGCOLOR()`) or from *palette* or from somewhere else. How would you transition it such case?
IMO not without effect layering and then blend two layers according to transition mode. Each layer would need to be painted separately with original and transitioned colors/palettes.